lib/config-validator.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { struct } = require('superstruct');
+
+const SolrError = require('./solr-error');
+
+const configStruct = struct.partial({
+	url: 'string',
+	core: 'string',
+	user: 'string?',
+	password: 'string?'
+});
+
+class ConfigValidator {
+
+	/**
+	 * Validates the received config struct
+	 * @param {Object} config the config to validate
+	 * @returns {Object} received config object
+	 * @throws if the struct is invalid
+	 */
+	static validate(config) {
+
+		try {
+
+			const validConfig = configStruct(config);
+
+			if((validConfig.user && !validConfig.password) || (validConfig.password && !validConfig.user))
+				throw new Error(`Password required for user '${validConfig.user}'`);
```

Comment:
En caso de que llegue `password` pero no `user`, devuelve: **Password required for user 'undefined'**
En el ticket solo se pedia que se valide `password` en caso de que llegue `user`, si no llega `user` se ignora la autenticación